core/server/services/bulk-email/index.js

```diff
@@ -46,24 +47,50 @@ module.exports = {
      * @param {Email} message - The message to send
      * @param {[EmailAddress]} recipients - the recipients to send the email to
      * @param {[object]} recipientData - list of data keyed by email to inject into the email
-     * @returns {Promise<boolean>} A promise representing the success of the email sending
+     * @returns {Array{Promise<boolean>}} An array of promises representing the success of the batch email sending
      */
     async send(message, recipients, recipientData) {
+        let BATCH_SIZE = 1000;
+
         if (!mailgunInstance) {
             return;
         }
         let fromAddress = message.from;
-        if (/@localhost$/.test(message.from)) {
+        if (/@localhost$/.test(message.from) || /@ghost.local$/.test(message.from)) {
             fromAddress = 'localhost@example.com';
             common.logging.warn(`Rewriting bulk email from address ${message.from} to ${fromAddress}`);
+
+            BATCH_SIZE = 2;
         }
         try {
-            const messageData = Object.assign({}, message, {
-                to: recipients.join(', '),
```

Comment:
There was no need to do this join as the SDK itself does the magic - https://github.com/mailgun/mailgun-js/blob/b88e942efa4c25f79606795a0536f665c4b173cb/lib/parse.js#L12. Less code on our side - easier to read :)

Reply:
very nice :relaxed: